proposals/0014-wave-size-range.md

```diff
@@ -0,0 +1,283 @@
+# Wave Size Range
+
+* Proposal: [0014](0014-wave-size-range.md)
+* Author(s): [Greg Roth](https://github.com/pow2clk)
+* Sponsor: [Greg Roth](https://github.com/pow2clk)
+* Status: **Under Consideration**
+* Planned Version: Shader Model 6.8, validator version 1.8
+
+## Introduction
+
+Shader Model 6.6 included the ability to specify a wave size on a shader entry
+ point in order to indicate that this shader depends on a specific wave size.
+If that size of wave isn't available to the current hardware,
```

Comment:
I think a key motivation of the SM 6.6 feature was that some hardware supports multiple wave sizes.  The attribute is required to force the driver/hardware to run that shader using the specified wave size.  Without it there's no way for the developer to tell the driver what the required wave size is.

So the feature wasn't really about allowing developers to choose which shader to load, it was about allowing developers to influence decisions that the driver was making when executing the shader.

Reply:
It was also about being able to write a shader that requires a particular wave size in an explicit way, such that it won't be allowed to run with a different wave size, since doing so can result in undefined behavior.